### PR TITLE
[TS] LPS-161170 Forms not editable after upgrade

### PR DESCRIPTION
**Pull request by IstvanD**
Dear Team,

Could you please review this pull request?

Thank you and best regards,
István

### CODE REVIEW

**Comment by rodrigopaulino** [comment]
ci:test:sf

**Comment by rodrigopaulino** [comment]
ci:test:relevant

**Comment by rodrigopaulino** [comment]
ci:test:upgrade

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_5/DDMStructureLayoutUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+					"DDMStructureVersion.structureId where ",
+					"DDMStructure.classNameId = ?"));
+			PreparedStatement preparedStatement2 =
+				AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+					connection,
+					"update DDMStructureLayout set definition = ? where " +
+						"structureLayoutId = ?")) {
+
+			preparedStatement1.setLong(
+				1, PortalUtil.getClassNameId(DDMFormInstance.class.getName()));
```
Comment (rodrigopaulino) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_5/DDMStructureLayoutUpgradeProcess.java:69]: Does this only happen with Forms? Have you checked if it's possible to have Web Content Structures or Document Types with the same case?

**Review by rodrigopaulino — CHANGES_REQUESTED** [review]
Hey, @IstvanD. Nice work!
I just have one question and one suggestion.
Thanks!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_5/DDMStructureLayoutUpgradeProcess.java (hunk 1 above)
Comment (IstvanD) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_5/DDMStructureLayoutUpgradeProcess.java:69]: Hi @rodrigopaulino I have checked it, but Web Content Structures and Document Types cannot be multi paged.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_5/DDMStructureLayoutUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+			try (ResultSet resultSet = preparedStatement1.executeQuery()) {
+				while (resultSet.next()) {
+					DDMFormLayout ddmFormLayout =
+						DDMFormLayoutDeserializeUtil.deserialize(
```
[the remark this answers is not in the document]
Comment (IstvanD) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_5/DDMStructureLayoutUpgradeProcess.java]: Hi @rodrigopaulino I have force-pushed the changes.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_5/DDMStructureLayoutUpgradeProcess.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+						continue;
+					}
+
+					definitionJSONObject.put(
+						"paginationMode", DDMFormLayout.MULTI_PAGES);
+
+					preparedStatement2.setString(
+						1, definitionJSONObject.toString());
+
+					long structureLayoutId = resultSet.getLong(
```
Comment (rodrigopaulino) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_5/DDMStructureLayoutUpgradeProcess.java:80]: You don't need this variable. You can put it directly as argument of the setLong.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_5/DDMStructureLayoutUpgradeProcess.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+					preparedStatement2.addBatch();
+				}
+
+				preparedStatement2.executeBatch();
+			}
+		}
+	}
+
+	private static final String _PAGINATED = "paginated";
```
Comment (rodrigopaulino) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_5/DDMStructureLayoutUpgradeProcess.java:93]: You don't need this constant. You can set the literal.

**Review by rodrigopaulino — CHANGES_REQUESTED** [review]
Thanks, @IstvanD. Looks nice. Just two minor tweaks and we're good to go!

**Comment by IstvanD** [comment]
> Thanks, @IstvanD. Looks nice. Just two minor tweaks and we're good to go!

Hi @rodrigopaulino got it! As before, I have force-pushed the changes.

**Review by rodrigopaulino — APPROVED** [review]
LGTM! 🎉 
Thanks a lot for your contribution, @IstvanD!

**Comment by rodrigopaulino** [comment]
ci:forward

**Comment by rodrigopaulino** [comment]
ci:reevaluate:1267401_4758

**Comment by rodrigopaulino** [comment]
ci:forward

**Comment by rodrigopaulino** [comment]
ci:forward

**Pull request closed without merging**